src/backend/distributed/sql/udfs/citus_mark_object_distributed/latest.sql

```diff
@@ -0,0 +1,7 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_mark_object_distributed(classId Oid, objectName text, objectId Oid)
```

Comment:
I think allowing regular users to call this function is a recipe for security problems. e.g. users with very low permissions could mark objects distributed that are not. Or could mark things as distributed that don't even exist. I'm not sure what happens in those cases, but I'm pretty sure lots of stuff will break (like adding new nodes). And having a user with very low permissions cause that kind of breakage is not a good thing.

I think we might want to have these connections to localhost to the main db use superuser instead, so that we can have this function (and probably some of the others), be executable by superuser only. This does bring some other security concerns though. We must make sure we're not executing anything as superuser that's not strictly needed. The commands in the main_db on the are harmless afaict (assuming we escape the arguments correctly). However, we'd want to execute the actual commands on the workers still as the user that the outer transaction uses. So we'd have to pass that along somehow. The easiest way seems to me to pass it as a second argument to either `execute_command_on_other_nodes` or to `citus_internal_start_management_transaction`.